components/certificate-mgt/org.wso2.carbon.identity.certificate.management.core/src/main/java/org/wso2/carbon/identity/certificate/management/core/ApplicationCertificateManagementServiceImpl.java

```diff
@@ -0,0 +1,196 @@
+import org.wso2.carbon.identity.certificate.management.core.util.CertificateMgtUtil;
+import org.wso2.carbon.identity.certificate.management.core.util.CertificateValidator;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+
+import java.util.UUID;
+
+/**
+ * This class is responsible for managing certificates of applications of a tenant.
+ * @deprecated Use {@link CertificateManagementServiceImpl} instead.
+ */
+public class ApplicationCertificateManagementServiceImpl implements ApplicationCertificateManagementService {
+
+    private static final Log LOG = LogFactory.getLog(ApplicationCertificateManagementServiceImpl.class);
+    private static final ApplicationCertificateManagementService INSTANCE =
+            new ApplicationCertificateManagementServiceImpl();
+    private static final ApplicationCertificateManagementDAOImpl DAO = new ApplicationCertificateManagementDAOImpl();
+    private static final CertificateValidator CERTIFICATE_VALIDATOR = new CertificateValidator();
+
+    private ApplicationCertificateManagementServiceImpl() {
+    }
+
+    public static ApplicationCertificateManagementService getInstance() {
+
+        return INSTANCE;
+    }
+
+    /**
+     * Add a certificate.
+     *
+     * @param certificate  Certificate information.
+     * @param tenantDomain Tenant domain.
+     * @return Auto incremented integer id of the Certificate.
+     * @throws CertificateMgtException If an error occurs while adding the certificate.
+     */
+    @Override
+    @Deprecated
+    public int addCertificate(Certificate certificate, String tenantDomain) throws CertificateMgtException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Adding certificate with name: " + certificate.getName());
+        }
+        doPreAddCertificateValidations(certificate);
+        String generatedCertificateId = UUID.randomUUID().toString();
+        return DAO.addCertificate(generatedCertificateId, certificate, IdentityTenantUtil.getTenantId(tenantDomain));
+    }
+
+    /**
+     * Get certificate information with given id.
+     *
+     * @param certificateId Certificate ID.
+     * @param tenantDomain  Tenant domain.
+     * @return Certificate information.
+     * @throws CertificateMgtException If an error occurs while getting the certificate.
+     */
+    @Override
+    @Deprecated
+    public Certificate getCertificate(int certificateId, String tenantDomain) throws CertificateMgtException {
+
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("Retrieving certificate with id: " + certificateId);
+        }
+        Certificate certificate = DAO.getCertificate(certificateId, IdentityTenantUtil.getTenantId(tenantDomain));
+        if (certificate == null) {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("No certificate found for the id: " + certificateId);
+            }
+            throw CertificateMgtUtil.raiseClientException(
```

Comment:
Rather than having a certificatemgtutil why not we have a CerticateManagementExceptionHanlder or a CerticateManagementExceptionManager

Reply:
shall we say throwClientException ?

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/6043/commits/35e940a974f67ca9f12296ced5a13a0e8bb9b444